app/helpers/events_helper.rb

```diff
@@ -149,8 +149,10 @@ def show_run_position(er, hover)
     elsif er.appliable and not er.assigned? and can? :create, er.applications.build(member: current_member)
       return link_to("you?", new_application_url(er.event, event_role_id: er.id, format: :js), :remote => true) unless hover
       "you?" if hover
+    elsif !hover and current_member and current_member.prefers_full_name
+      return er.assigned_to :both_names
     elsif !hover and current_member
-      er.assigned_to use_display_name: true
+      return er.assigned_to :display_name
     else
       er.assigned_to
     end
```

Comment:
Can we use `run_position_name_preference()` here too to be consistent? Maybe:
```ruby
elsif !hover and current_member
  return er.assigned_to run_position_name_preference(current_member)
```
That way if we add more options or change things in the future it lowers the number of places in the code to update.